app/docs/enterprise/0.31-x/kong-architecture-patterns.md

```diff
@@ -0,0 +1,60 @@
+
+## Kong and APIs behind firewall
+
+The next pattern we see, and the next step for security, is to setup [API servers behind a firewall](https://getkong.org/docs/latest/network/) and only allow the Kong API Gateway be accessible to the public.
+
+This is a popular architecture pattern that we see with customers who have publicly available APIs. Either they are migrating from the initial pattern, improving security by cordoning off direct access to the APIs, or choosing it for their general use. 
+
+This architecture pattern is the one we recommend for most customers. It is straightforward to setup, and offers security and control over API access.
+
+The Kong cluster would be placed behind a firewall, or within an Amazon VPC. Port 80 would be opened to the outside world for just the Kong cluster. Inside the firewall the API servers would only have allowed connections from the Kong cluster. Other IP:ports may need to be opened for the API servers to reach resources they need to fulfill any requests.
+
+
+Access to the [Kong Admin API can be secured in a couple of different ways](https://getkong.org/docs/latest/secure-admin-api/). Depending on your needs you would lock it down to localhost on the machine it is running on, use the firewall to prevent access from anyone outside your network, or proxy the Admin API through Kong itself, and secure it using one of the authentication plugins.
+
+With Kong Enterprise you also have the option of using Role Based Access Control (RBAC) to shape fine grained access for your team, and co-workers. [RBAC in Kong](https://getkong.org/docs/enterprise/0.31-x/plugins/rbac-api/) is used to create roles with defined permissions; roles are then assigned to users who are limited in what they can and can not do by the permissions. Roles could be created for read-only roles; for when you want to let co-workers look, but not edit, the configuration. You could also create roles for different teams; giving them access to edit and update only their APIs.
+
+## Multi-datacenter with Cassandra
+
+Application availability is important, and high availability can be achieved by configuring Kong in a multi-datacenter setup using [Apache Cassandra](https://getkong.org/about/faq/#apache-cassandra) as the backing datastore. In this configuration each geographically disparate datacenter is host to at least three Kong nodes, and the appropriate number of Cassandra nodes.
+
+The minimum number of Kong nodes per data center needed for high availability is three. This will ensure that in the unlikely event of a node failure there will still be a node and a backup. Cassandra would be configured in an optimal way; you can use the [Cassandra Calculator](https://www.ecyrd.com/cassandracalculator/) to determine how many Cassandra nodes are needed per data center to ensure high availability, and  performance for your specific needs.
```

Comment:
I would reword the sentence "Casandra would be configured in an optimal way;"  Maybe something like "_To ensure Casandra is configured int the an optimal way we recommend using the [Casandra Calculator]_"

Reply:
Noted.

Reply:
"and  performance" has 2 spaces